pkg/archive/tar.go

```diff
@@ -157,6 +157,26 @@ func Apply(ctx context.Context, root string, r io.Reader, opts ...ApplyOpt) (int
 	return options.applyFunc(ctx, root, r, options)
 }
 
+// CachedRootPath is a cache for root paths, designed to optimize performance by
+// eliminating redundant calculations. This is particularly beneficial when
+// dealing with folders containing numerous files, as each call to RootPath
+// would otherwise require re-traversing the directory structure, which is
+// unnecessary after the first call.
+type CachedRootPath map[string]string
+
+func (c CachedRootPath) Get(root, path string) (string, error) {
+	key := root + path
```

Comment:
I think we need to add a separator between root and path

Otherwise, e.g., "foo" + "bar" and "foob" + "ar" will map to the same key.

Reply:
oh, right, adding